sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -60,16 +60,24 @@ public override AuthenticationToken GetToken(GetTokenOptions properties, Cancell
             // Check if scopes are present and in a valid format
             if (properties.TryGetValue(GetTokenOptions.ScopesPropertyName, out var scopesValue))
             {
-                if (scopesValue is ReadOnlyMemory<string> scopes)
+                if (scopesValue is ReadOnlyMemory<string> readOnlyMemoryScopes)
                 {
                     return new GetTokenOptions(properties);
                 }
-                else
+                // Try to convert scopes to ReadOnlyMemory<string>
+                else if (scopesValue is string[] stringArrayScopes)
                 {
-                    throw new ArgumentException($"If a \"{GetTokenOptions.ScopesPropertyName}\" property is included in the properties, it should be typed as ReadOnlyMemory<string>.", nameof(properties));
+                    ReadOnlyMemory<string> scopes = new ReadOnlyMemory<string>(stringArrayScopes);
+                    // Create new properties dictionary with properly formatted scopes
```

Comment:
why do you need this code below? Why not call return new GetTokenOptions(scopes); at this point?

Reply:
We could optimize and only create the new dictionary if there are more.